client/lib/html-toolbar/index.js

```diff
@@ -0,0 +1,228 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { Env } from 'tinymce/tinymce';
+import { noop, reduce } from 'lodash';
+
+/**
+ * Return an object containing the textarea content split into:
+ * - `before` selection
+ * - (`inner`) selection
+ * - `after` selection
+ *
+ * @param {Element} textarea The editor textarea.
+ * @returns {Object} The split textarea content.
+ */
+export const splitSelectedContent = textarea => {
+	const { selectionEnd, selectionStart, value } = textarea;
+	return {
+		before: value.substring( 0, selectionStart ),
+		inner: value.substring( selectionStart, selectionEnd ),
+		after: value.substring( selectionEnd, value.length ),
+	};
+};
+
+/**
+ * Set the cursor position after the insertion of new content into a textarea.
+ *
+ * @param {Element} textarea The editor textarea.
+ * @param {Number} selectionEnd The textarea.selectionEnd value BEFORE the content insertion.
+ * @param {Number} insertedContentLength The length of the inserted content.
+ */
+export const setCursorPosition = ( textarea, selectionEnd, insertedContentLength ) => {
+	textarea.selectionEnd = textarea.selectionStart = selectionEnd + insertedContentLength;
+};
+
+/**
+ * Insert content into a textarea.
+ *
+ * @param {Element} textarea The target textarea.
+ * @param {String} before The content before the selection.
+ * @param {String} content The new content to insert.
+ * @param {String} after The content after the selection.
+ * @param {Function} onInsert A function to call after the insertion.
+ */
+export const insertContent = ( textarea, before, content, after, onInsert = noop ) => {
+	const { selectionEnd, value } = textarea;
+	const newContent = before + content + after;
+
+	if ( Env.gecko ) {
```

Comment:
Let's avoid pulling in Env, and use `document.queryCommandSupported` instead. As a general note feature detection over browser detection is preferred.

https://developer.mozilla.org/en-US/docs/Web/API/Document/queryCommandSupported

Reply:
Shoot!
I forgot we had split the comments bundle!
I'll see to sniff in another way. :)

Reply:
@gwwar oooooh, great reminder <3

Reply:
<img width="840" alt="screen shot 2018-02-08 at 10 41 20 am" src="https://user-images.githubusercontent.com/1270189/35991700-da2bd9d6-0cbc-11e8-8d22-b1b10576751a.png">

Hmm, still a 50% jump, but much better than +500%. Let's 👀 if this looks okay